cartridges/int_algolia/cartridge/scripts/algolia/lib/algoliaSplitter.js

```diff
@@ -61,11 +63,18 @@ function splitHtmlContent(htmlContent, maxByteSize, splitterElement) {
  * @returns {string} Sanitized content
  */
 function removeIgnoredContent(content) {
-    //remove restricted tags and their content
     IGNORED_TAGS.forEach(function(tag) {
-        content = content.replace(new RegExp('<' + tag + '.*?' + tag + '>', 'g'), '');
+        // Remove entire <tag ...>...</tag> blocks (multiline)
+        content = content.replace(
+            new RegExp('<' + tag + '[^>]*>[\\s\\S]*?<\\/' + tag + '>', 'gi'),
+            ''
+        );
+        // Remove self-closing tags <tag ... />
```

Comment:
Good, I think this addresses this comment thanks: https://github.com/algolia/algoliasearch-sfcc-b2c/pull/164/files#r1565893249

Please add unit tests on the `removeIgnoredContent` function directly to confirm, including tests with self-closing tags.